keps/draft-20180426-security-conformance.md

```diff
@@ -0,0 +1,682 @@
+  and possibly offering better compatibility and capabilities.
+  For example, using hypervisors, gVisor etc. which provides an additional layer of protection.
+
+Note: a workload is possible to be compromised if it's not properly
+designed/implemented according to security standards, or it's vulnerable.
+A compromised user identity can be used to compromise the whole system.
+These are out-of-scope of this proposal.
+
+The criteria described below is based on a namespace with default settings.
+Like a newly created namespace, with only service account `default`.
+Namespaces with special/privileged settings, like `kube-system`,
+or special namespaces required by some system components
+(e.g. nginx ingress controller) are not covered.
+
+Additional profiles can be proposed with combination of existing and new criteria.
+
+### Criteria
+
+#### Config File Permissions
+
+  *Conformance Profile: Essential*
+
+  The configuration files for API server, controller manager, scheduler and kubelet should have the correct setup of ownership and file permissions.
+  Without proper permission setup, a local user will be able to access prohibited information, including
+  - private keys
+  - secrets
+  - credentials
+
+  The criteria are:
+  - etcd data directory is owned and accessible only by the user running etcd;
+  - kubeconfig files for control plane is owned and accessible only by the user runs the control plane;
+  - private key files is owned and accessible only by the user who runs the process which consumes the key;
+  - kubeconfig for administration purpose (if present, e.g. `/etc/kubernetes/admin.conf`)
+    is owned and accessible only by the admin user (perm 0400)/group (perm 0440)
+    (the test doesn't verify particular user/group, but only the permissions).
+
+
+  Notes:
+  - the above statements exclude _root_ user who always have full access to the system.
+  - if etcd is managed separately, the test may not cover it.
+
+
+##### Test methodology
+
+  These tests will require the accessibility to the host.
+  There are a few options to achieve that:
+
+    - Create a Pod with special privilege (e.g. in `kube-system` namespace) and
+    run with host pid namespaces, bind mount host file system into testing container for inspection.
+    E.g. creating a Pod using DaemonSet.
+    This is the most consistent way with other tests without requiring user to perform additional operations before running the tests.
+
+    - Remote access to master, like SSH.
+    This is the alternative way if it's impossible to schedule a Pod on master (the kubelet on master doesn't register).
+
+  The test code will inspect the master/node system by first looking up the processes by names.
+  The names are default to `etcd`, `kube-apiserver`, `kube-controller-manager`, `kube-scheduler`, `kubelet`.
+  In the case when the cluster admin changes these names,
+  overriding mechanism (e.g. via environment variable) must be supported by test code.
```

Comment:
I'm worried you're going to end up needing to define the full cluster API to fully configure these tests...